src/db.c

```diff
@@ -40,6 +40,7 @@ typedef enum {
 } keyStatus;
 
 static keyStatus expireIfNeeded(redisDb *db, robj *key, kvobj *kv, int flags);
+static keyStatus expireIfNeededWithKV(redisDb *db, kvobj *kv, int flags);
```

Comment:
I prefer to extend `expireIfNeeded`, let it support key is none, that is, `key` also is an option like `kv`. And `keyIsExpired` already supports this usage. 
```
keyStatus expireIfNeeded(redisDb *db, robj *key, kvobj *kv, int flags) {
    serverAssert(key != NULL || kv != NULL);
    sds keyname = key ? key->ptr : NULL;
    ...
    /* Delete the key */
    if (key) {
        deleteExpiredKeyAndPropagate(db, key);
    } else {
        sds keyname = kvobjGetKey(kv);
        robj *tmpkey = createStringObject(keyname, sdslen(keyname));
        deleteExpiredKeyAndPropagate(db, tmpkey);
        decrRefCount(tmpkey);
    }
```
this approach can avoid introducing this new function that is a bit duplicated.

cc @moticless could you share your thoughts?

Reply:
Agree. Note also that `keyIsExpiredWithKV` is also redundant. The function `keyIsExpired` is gracefull to accept either the keyname or kv for the lookup.